geonode/static/geonode/js/upload/upload.js

```diff
@@ -352,6 +352,10 @@ define(['underscore',
             buildFileInfo(_.groupBy(file_input.files, path.getName));
             displayFiles(file_queue);
         });
+        // Detect click on "Remove" link and update the file_queue
+        $(options.file_queue).on('click', 'a', function () {
+            displayFiles(file_queue);
+        });
```

Comment:
Yeah, that is a good point. I agree with your thinking. I believe I should just have to modify the generated element to have a class attribute, then I can do that. I'll get that update in tomorrow morning for me, or if it's not possible for some reason I'll post back here with the problem.